File: src/losange.py
class Losange:
    def __init__(self, diagonale1, diagonale2):
        if diagonale1 <= 0 or diagonale2 <= 0:
            raise ValueError("Les diagonales doivent être positives")

        self.diagonale1 = diagonale1
        self.diagonale2 = diagonale2

    def get_area(self):
        return (self.diagonale1 * self.diagonale2) / 2

File: src/losange.js
class Losange {
  constructor(diagonale1, diagonale2) {
    if (diagonale1 <= 0 || diagonale2 <= 0) {
      throw new Error("Les diagonales doivent être positives");
    }

    this.diagonale1 = diagonale1;
    this.diagonale2 = diagonale2;
  }

  getArea() {
    return (this.diagonale1 * this.diagonale2) / 2;
  }
}

module.exports = Losange;